### PR TITLE
Add integration test to test against latest skaffold release

### PR DESCRIPTION
**Pull request by priyawadhwa**
This PR:

1. Adds an integration test to test minikube against the latest skaffold release
1. Adds in a sample skaffold app from the skaffold repo to test
1. Adds this test to the addons_certs_docker_ubuntu Github Actions test

Fixes https://github.com/kubernetes/minikube/issues/8752

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -612,7 +612,7 @@ jobs:
           chmod a+x e2e-*
           chmod a+x minikube-*
           START_TIME=$(date -u +%s)
-          KUBECONFIG=$(pwd)/testhome/kubeconfig MINIKUBE_HOME=$(pwd)/testhome ./e2e-linux-amd64 -minikube-start-args=--driver=docker  -test.run "(TestAddons|TestCertOptions)" -test.timeout=10m -test.v -timeout-multiplier=1.5 -binary=./minikube-linux-amd64 2>&1 | tee ./report/testout.txt
```
Comment (medyagh) [inline .github/workflows/pr.yml:615]: add same s|TestSkaffold to all addon tests in all platforms (virtualbox, docker ... ) there are multiple jobs

and also then copy the whole thing except the first 10 lines to master.yml

File: test/integration/skaffold_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	"os/exec"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-getter"
+	"k8s.io/minikube/pkg/util/retry"
+)
+
+func TestSkaffold(t *testing.T) {
```
Comment (medyagh) [inline test/integration/skaffold_test.go:35]: how about to call this test, TestExternalTools

and make skaffold a subtest of this test ?

Comment (priyawadhwa) [reply test/integration/skaffold_test.go:35]: do we have other external tools we plan on testing?

File: .github/workflows/pr.yml (hunk 1 above)
Comment (priyawadhwa) [reply .github/workflows/pr.yml:615]: thanks, done!

**Comment by priyawadhwa** [comment]
Hey @medyagh thank you! This should be RFAL.

**Comment by medyagh** [comment]
Docker driver skaffold test fialed

	creating runner: creating builder: getting docker client: getting minikube env: running [minikube docker-env --shell none]
	 - stdout: "================================================================================\nAn error has occurred. Would you like to opt in to sending anonymized crash\ninformation to minikube to help prevent future errors?\nTo opt out of these messages, run the command:\n\tminikube config set WantReportErrorPrompt false\n================================================================================\n"
	 - stderr: "There is a newer version of minikube available (v1.12.1).  Download it here:\nhttps://github.com/kubernetes/minikube/releases/tag/v1.12.1\n\nTo disable this notification, run the following:\nminikube config set WantUpdateNotification false\nE0721 16:00:08.316805   27065 env.go:

**Comment by priyawadhwa** [comment]
@medyagh thanks, the test is fixed

File: test/integration/skaffold_test.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func TestSkaffold(t *testing.T) {
 	if err != nil {
 		t.Fatalf("absolute path to minikube binary: %v", err)
 	}
-	os.Setenv("PATH", fmt.Sprintf("%s:%s", os.Getenv("PATH"), filepath.Dir(abs)))
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", filepath.Dir(abs), os.Getenv("PATH")))
```
Comment (medyagh) [inline test/integration/skaffold_test.go:70]: this probably won't work in windows ?

File: test/integration/skaffold_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	// copy minikube binary to minikube
+	if err := mcnutils.CopyFile(Target(), filepath.Join(filepath.Dir(abs), "minikube")); err != nil {
+		t.Fatalf("error copying to minikube")
+	}
+	if err != nil {
+		t.Fatalf("absolute path to minikube binary: %v", err)
+	}
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", filepath.Dir(abs), os.Getenv("PATH")))
+	// make sure 'docker' and 'minikube' are on PATH
+	for _, binary := range []string{"minikube", "docker"} {
```
Comment (medyagh) [inline test/integration/skaffold_test.go:72]: the minikube binary in our tests is not called minikube, but minikube-os-arch
and some of our CI machines we have a very old minikube installed in the /usr/loca/bin/
this means skaffold will use that binary?

File: test/integration/skaffold_test.go (hunk 3 above)
Comment (priyawadhwa) [reply test/integration/skaffold_test.go:70]: why won't this work on windows?

File: test/integration/skaffold_test.go (hunk 4 above)
Comment (priyawadhwa) [reply test/integration/skaffold_test.go:72]: minikube-GOOS-amd64 is copied to`minikube` in line 64 of this test

the `minikube` directory is added to the beginning of PATH in line 70 so we know that skaffold will use the correct binary

**Pull request merged**